### DISCUSSION TITLE
Umi.js chunk shows application code

### DISCUSSION THREAD

**Discussion by techwizard-prasad** [umijs/umi#10665] [Q&A]
When I try to analyze the build, I found out the umi.js chunk includes part of application code as well. With Parsed size of 800KB+.
I am trying to reduce the size of UMMI chunk. `splitChunks` is not helping either.
I am attaching the analyze response below. Any other tip to reduce this chunk size would be really helpful.
Also, I dont understamd why my application code is being shipped with umi js
Thank you in advance for the help.
![image](https://user-images.githubusercontent.com/53524689/222442636-7e9b6104-bce9-42a5-80be-4f64c829e736.png)

**Comment by fz6m** [comment]
Using a split chunk strategy ( Umi4 only ):

```ts
// .umirc.ts
export default {
  codeSplitting: {
    jsStrategy: 'granularChunks'
  }
}
```

Or manually split chunk :

```tsx
import { lazy, Suspense } from 'react'
 
// split `./Page` chunk
const Page = lazy(() => import('./Page'))
 
export default function() {
  return (
    <Suspense fallback={<div>loading...</div}>
      <Page />
    </Suspense>
  )
}
```

**Reply by techwizard-prasad** [reply]
Thanks @fz6m 
But I am using UMI3. Is there any way we can do the same in V3?
